### PR TITLE
Reduced muon validation for PhaseII Upgrade high PU scenarios (90X)

### PR DESCRIPTION
**Pull request by calabria**
@kpedro88

### CODE REVIEW

**Comment by kpedro88** [comment]
@cmsbuild please test

**Comment by civanch** [comment]
please test

**Comment by civanch** [comment]
+1

**Comment by slava77** [comment]
@cmsbuild please test

**Comment by kpedro88** [comment]
please abort

**Comment by kpedro88** [comment]
please test

**Comment by civanch** [comment]
+1

**Comment by slava77** [comment]
@cmsbuild please test

**Comment by civanch** [comment]
+1

**Comment by dmitrijus** [comment]
+1

**Comment by slava77** [comment]
+1

for https://github.com/cms-sw/cmssw/pull/16669 5029b88
- code changes are as intended
- jenkins tests pass and comparisons show differences in 2023 wflows in muon isolation and calorimeter quantities (expected)

See more notes on technical performance in https://github.com/cms-sw/cmssw/pull/16670#issuecomment-263006048

**Comment by Martin-Grunewald** [comment]
+1

**Comment by slava77** [comment]
@davidlange6 
are there any issues with this PR or is it just to early to expect it to be merged?

File: HLTrigger/Configuration/python/customizeHLTforCMSSW.py [hunk 1]
```diff
@@ -58,6 +58,13 @@ def customiseFor14833(process):
                 producer.includeME0 = cms.bool(False)
     return process
 
+def customiseFor16670(process):
```
Comment (davidlange6) [inline HLTrigger/Configuration/python/customizeHLTforCMSSW.py]: Hi @calabria @slava77 -does not a fillDescriptions implementation avoid this function completely? In any case, when will this get propagated to the menu?

Comment (fwyzard) [inline HLTrigger/Configuration/python/customizeHLTforCMSSW.py]: we plan to migrate the HLT menu to CMSSW 9.0.x once pre1 is out.

Comment (davidlange6) [inline HLTrigger/Configuration/python/customizeHLTforCMSSW.py]: any reason a fillDescription does not work here? (with the nice feature of being desirable for the long term)

Comment (slava77) [inline HLTrigger/Configuration/python/customizeHLTforCMSSW.py]: it's a typical problem of no easy fix.
There is not fillDescriptions for this tool.

It's an ESProducer, it currently gets a tool to be configured from the plugin factory during ::produce call, the needed parameter is a sub-tool of the tool to be configured.
So, some significant development and some refactoring might be necessary.

@Dr15Jones do we have an example of what to do in this case?

**Comment by davidlange6** [comment]
I don’t follow the complexity (yet) It looks like ESSources support fillDescriptions. Is the the HLT config not a standalone ESSource as it is in 
TrackingTools/TrackAssociator/python/DetIdAssociatorESProducer_cff.py - but ok, I haven’t tried this myself...

in fact, is this parameter in the menu at all? I don’t really see it aside from here
https://github.com/cms-sw/cmssw/blob/53327033fb356b2a704464439ce79fa47a9d1678/HLTrigger/HLTanalyzers/python/HLT_ES_cff.py

is this somehow derived from online menus or editable in the release (unedited since <62x)

**Comment by Martin-Grunewald** [comment]
AQt HLT we have currently exactly 5 parameters for any ```DetIdAssociatorESProducer``` instance
in the menu, even 
```
process.muonDetIdAssociator = cms.ESProducer( "DetIdAssociatorESProducer",
  ComponentName = cms.string( "MuonDetIdAssociator" ),
  etaBinSize = cms.double( 0.125 ),
  nEta = cms.int32( 48 ),
  nPhi = cms.int32( 48 ),
  includeBadChambers = cms.bool( False )
)
```
So any missing parameters must either already be inserted by fillDescriptions, or via existsAs treated in the code.

**Comment by Dr15Jones** [comment]
@slava77 to my knowledge, we've never come up with a satisfactory mechanism of dealing with configuration parameters from internal plugins of modules.

**Comment by fwyzard** [comment]
Indeed.

Would it make sense to add a static `fillDescriptions`-like method to the classes that make use of the plugin system ?

**Comment by slava77** [comment]
@davidlange6 
would existsAs solution be acceptable to you?
(it was suggested by @Martin-Grunewald )
This is simple enough that I can see it implemented quickly.

**Comment by fwyzard** [comment]
I'd rather we finally have a general way of handling plugins, say in a month or so, and live with a python customisation in the meantime.

**Comment by davidlange6** [comment]
+1

**Comment by Dr15Jones** [comment]
@fwyzard I don't have any plans for a 'general way' of handling plugins. The difficulty is the validator object created in the `fillDescriptions` method of the module would have to know how to load the plugin itself (talking to a different interface connected to the plugin's fillDescription rather than the regular plugin interface) and then know what parameters from the module must be passed to the plugin.

The extremely wide diversity of how plugins are configured (on top of the pervasiveness of plugins) makes a general solution extremely difficult.

**Pull request merged**